forte/data/data_store.py

```diff
@@ -0,0 +1,239 @@
+        r"""This function adds an annotation entry with `begin` and `end` index
+        to the sortedlist at index `type_id` of the array which records all
+        sortedlists, return tid for the entry.
+
+        Args:
+            type_id (int): The index of Annotation sortedlist in the array.
+            begin (int): begin index of the entry.
+            end (int): end index of the entry.
+
+        Returns:
+            The tid of the entry.
+        """
+        # We should create the entry list with the format
+        # [entry_type, tid, begin, end, None, ...].
+        # A helper function _new_entry() can be used to generate a entry list
+        # with default fields.
+        # A reference to the entry should be store in both self.elements and
+        # self.entry_dict.
+        raise NotImplementedError
+
+    def set_attr(self, tid: int, attr_name: str, attr_value):
+        r"""This function locates the entry list with tid and sets its
+        attribute `attr_name` with value `attr_value`.
+
+        Args:
+            tid (int): Unique id of the entry.
+            attr_name (str): name of the attribute.
+            attr_value: value of the attribute.
+
+        """
+        # We retrieve the entry list from entry_dict using tid. We get its
+        # entry type. We then locate the index of the attribute using the entry
+        # type, field dictionary and attr_name, and update the attribute.
+
+        raise NotImplementedError
+
+    def get_attr(self, tid: int, attr_name: str):
+        r"""This function locates the entry list with tid and gets the value
+        of `attr_name` of this entry.
+
+        Args:
+            tid (int): Unique id of the entry.
+            attr_name (str): name of the attribute.
+
+        Returns:
+            The value of attr_name for the entry with tid.
+        """
+        # We retrieve the entry list from entry_dict using tid. We get its
+        # entry type. We then locate the index of the attribute using the entry
+        # type, field dictionary and attr_name, and get the attribute.
+
+        raise NotImplementedError
+
+    def delete_entry(self, tid: int):
+        r"""This function locates the entry list with tid and removes it
+        from the data store. It removes the entry list from both entry_dict
+        and the sortedlist of its type.
+
+        Args:
+            tid (int): Unique id of the entry.
+
+        """
+        # We retrieve the entry list from entry_dict using tid. We get its
+        # entry type, type id, begin and end indexes. Then, we find the
+        # `entry_type` sortedlist using type id. We bisect the sortedlist
+        # to find the entry list. We then remove the entry list from both
+        # entry_dict and the `entry_type` sortedlist.
+
+        raise NotImplementedError
+
+    def get_entry(self, tid: int) -> List:
+        r"""Look up the entry_dict with key `tid`.
```

Comment:
Here you should as well get the location information of this entry (for example, which sortedlist, and its index in the sortedlist). So that this info can be reuse later, in functions like `next_entry`